active-rfcs/0000-scoped-css-with-preserved-specificity.md

```diff
@@ -0,0 +1,146 @@
+|:------------------ |:----------------------------------------------------------------------------------- |:--------------------:|:--------------------:|
+| `p`                | `p:where(.astro-OZV3B5RX)`                                            | `0.0.1`              | `0.0.1`              |
+| `aside p`          | `aside:where(.astro-OZV3B5RX) p:where(.astro-OZV3B5RX)` | `0.0.2`              | `0.0.2`              |
+| `:global(aside) p` | `aside p:where(.astro-OZV3B5RX)`                                      | `0.0.2`              | `0.0.2`              |
+
+
+By using `:where()`, styles will now:
+
+- respect the specificity with which they were authored.
+- work consistently alongside other CSS files or CSS libraries.
+- still preserve the exclusive boundaries that prevent styles from applying outside the component.
+
+
+
+# Motivation
+
+Astro component styles are scoped by default, meaning they only apply to items in the component.
+These are _exclusive_ styling boundaries that prevent styles from applying to items outside the component or within child components.
+
+This specificity increase is perceptually inconsistent to authors, without a deeper knowledge of the Astro internals and **Type**, **ID**, add **Class** selector specificity.
+This specificity increase makes it hard to combine with other CSS files or other styling libraries (Tailwind, CSS Modules, Styled Components, Stitches), as those systems would not include our additional side-effects.
+
+From the earlier example, the HTML and CSS currently transforms into the following code:
+
+```astro
+<style>
+p.astro-OZV3B5RX {
+  color: blueviolet;
+}
+
+aside.astro-OZV3B5RX p.astro-OZV3B5RX {
+  color: darkviolet;
+}
+
+aside p.astro-OZV3B5RX {
+  text-decoration: underline;
+}
+</style>
+<p class="OZV3B5RX">This is a paragraph.</p>
+<aside class="OZV3B5RX">
+  <p class="OZV3B5RX">This is a paragraph in an aside.</p>
+</aside>
+```
+
+| Authored Selector  | Compiled Selector                                                   | Authored Specificity | Compiled Specificity |
+|:------------------ |:------------------------------------------------------------------- |:--------------------:|:--------------------:|
+| `p`                | `p.astro-OZV3B5RX`                                    | `0.0.1`              | `0.1.1`              |
+| `aside p`          | `aside.astro-OZV3B5RX p.astro-OZV3B5RX` | `0.0.2`              | `0.2.2`              |
+| `:global(aside) p` | `aside p.astro-OZV3B5RX`                              | `0.0.2`              | `0.1.2`              |
+
+
+
+# Detailed design
+
+Update the compiler to append `:where(.astro-XXXXXXXX)` rather than `.astro-XXXXXXXX`.
+
+```css
+/* Before (specificity of 0.1.1) */
+h1.astro-XXXXXXXX {}
+
+/* After (specificity of 0.0.1) */
+h1:where(.astro-XXXXXXXX) {}
+```
+
+
+
+# Drawbacks
```

Comment:
I think this needs some drawbacks to strength the case. There are definitely some. Some users might expect the styles they write within a component to "always win" and this would break that expectation.